### PR TITLE
web: add strong ssl_ciphers and headers

### PR DESCRIPTION
**Pull request by netaskd**
The commit improves security the web application.
https://www.ssllabs.com/ssltest gives A+ overall rating without "WEAK" ciphers.

### CODE REVIEW

**Comment by saghul** [comment]
Thank you! Can you share a result report? I want to see if all our supported systems (mobile, mostly) would still be supported.

File: web/rootfs/defaults/ssl.conf [hunk 1]
```diff
@@ -18,4 +18,11 @@ ssl_certificate_key /config/keys/cert.key;
 # protocols
 ssl_protocols TLSv1.2;
 ssl_prefer_server_ciphers on;
-ssl_ciphers ECDH+AESGCM:ECDH+CHACHA20:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:RSA+AESGCM:RSA+AES:!aNULL:!MD5:!DSS;
+ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!3DES:!MD5:!PSK;
+ssl_ecdh_curve secp384r1;
+
+# headers
+add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload";
+add_header X-Frame-Options DENY;
```
Comment (saghul) [inline web/rootfs/defaults/ssl.conf]: Won't this prevent the external API from functioning? That is, being embedded in an iframe.

File: web/rootfs/defaults/ssl.conf [hunk 2]
```diff
@@ -18,4 +18,11 @@ ssl_certificate_key /config/keys/cert.key;
 # protocols
 ssl_protocols TLSv1.2;
 ssl_prefer_server_ciphers on;
-ssl_ciphers ECDH+AESGCM:ECDH+CHACHA20:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:RSA+AESGCM:RSA+AES:!aNULL:!MD5:!DSS;
+ssl_ciphers ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!3DES:!MD5:!PSK;
```
Comment (saghul) [inline web/rootfs/defaults/ssl.conf]: Can't all ECDHE ciphers with GCM be collapsed into ECDHE+AESGCM? Also, why no ChaCha20?

File: web/rootfs/defaults/ssl.conf (hunk 1 above)
Comment (netaskd) [reply web/rootfs/defaults/ssl.conf]: Not sure. I don't have a chance to check it. Possibly can break.

File: web/rootfs/defaults/ssl.conf (hunk 2 above)
Comment (netaskd) [reply web/rootfs/defaults/ssl.conf]: It is parameters that I use in production. I don't deep dive to meaning each of them, and rely on ssllabs recommendations. About CHACHA20, missed this inclusion

**Comment by netaskd** [comment]
Here a report from ssl labs about the parameters (domain and ip changed):
[ssllabs_report.pdf](https://github.com/jitsi/docker-jitsi-meet/files/2649399/ssllabs_1.pdf)

File: web/rootfs/defaults/ssl.conf (hunk 1 above)
Comment (saghul) [reply web/rootfs/defaults/ssl.conf]: Yep, this breaks the external API apparently. Can you please drop it from this PR? I'm happy to discuss adding a setting for it.

**Comment by netaskd** [comment]
Hi @saghul,
Thanks for the testing X-Frame-Options! It's removed from PR.
Also I've added CHACHA20 ciphers. Now it looks like: cipther list

**Pull request merged**